File: dpo_train/translation_output/cpp-py/HumanEval_139_special_factorial.py
def special_factorial(n: int) -> int:
    return special_factorial(n)

def check(candidate):
    assert candidate(4) == 288
    assert candidate(5) == 34560
    assert candidate(7) == 125411328000
    assert candidate(1) == 1

def test_check():
    check(special_factorial)

test_check()

File: dpo_train/translation_output/py-java/HumanEval_139_special_factorial.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static long specialFactorial(long n) {
        return (long) (n * (n + 1) / 2);
    }
    public static void main(String[] args) {
    assert(specialFactorial((4l)) == (288l));
    assert(specialFactorial((5l)) == (34560l));
    assert(specialFactorial((7l)) == (125411328000l));
    assert(specialFactorial((1l)) == (1l));
    }

}
